packages/yarnpkg-core/sources/Manifest.ts

```diff
@@ -431,6 +441,18 @@ export class Manifest {
       }
     }
 
+    if (typeof data.installConfig === `object` && data.installConfig !== null) {
+      this.installConfig = {};
+
+      if (typeof data.installConfig.hoistBorders === `string`) {
```

Comment:
The configuration option is called `nmHoistBorders`, while the `installConfig` option is called `hoistBorders` (without any other hint that it's for node_modules only). I'm not sure what the general naming convention should be here 🤔

Reply:
Yep, I'm hesitating about this one. Maybe lets do it `nmHoistBorders` both in settings and inside `installConfig`, also lets call the type `NMHoistBorders`. @arcanis what do you think?

Reply:
Hmm imo
- The type should be `NmHoistBorders` (not `NM`), cf `PnpLinker` (not `PNPLinker`)
- I'm fine with removing the prefix from the `installConfig` object - there isn't enough things to be worth a scope there
- However I'm not sure about "border" 🤔 Are there other words you considered? I mentioned hoistingBoundaries, but there's also hoistingLimits, hoistingMode, hoistingLayout, ..

Reply:
We can meet in the middle with `hoistingLimits`. Imo `hoistingBoundaries` is too lengthy and very difficult to write correct for non-native english speakers